PY-SDK/public/src/pyaver/user_host_lifetime.py

```diff
@@ -264,6 +268,32 @@ def derive_pubkey_and_bump(owner: PublicKey, host: PublicKey, program_id: Public
             program_id
         )
 
+    async def make_update_user_host_lifetime_state_instruction(self):
+        program = await self.aver_client.get_program_from_program_id(self.program_id)
+        # TODO
+        return None
+
+    async def update_user_host_lifetime_state(self, fee_payer: Keypair = None, send_options: TxOpts = None):
+        if(fee_payer == None):
+            fee_payer = self.aver_client.owner
+
+        ix = await self.make_update_user_host_lifetime_state_instruction()
+
+        return await sign_and_send_transaction_instructions(
+            self.aver_client,
+            [],
+            fee_payer,
+            [ix],
+            send_options
+        )
+
+    async def check_if_uhl_latest_version(self):
+        program = await self.aver_client.get_program_from_program_id(self.program_id)
+        if(self.user_host_lifetime_state.version < get_version_of_account_type_in_program(AccountTypes.USER_HOST_LIFETIME, program)):
+            print("User host lifetime needs to be upgraded")
```

Comment:
I feel like Cian would want a more useful error message here for the user perhaps? Something which tells them why accounts needs to be upgraded and why